### PR TITLE
feat: allow await outside function

### LINKED ISSUE #188: Add support for `top-level await`
just noticed an Error in `warnings`:
```
Error: Failed to parse xxx.js as module:
Cannot use keyword 'await' outside an async function
    at Object.analyze [as default] ...
```

it already supported in node>=14.8.0, so i think there will be more and more projects using top-level await in the future.

### PR DESCRIPTION
**Pull request by hehex9**
closes #188

### CODE REVIEW

**Review by guybedford — APPROVED** [review]

File: src/analyze.ts [hunk 1]
```diff
@@ -239,7 +239,7 @@ export default async function analyze(id: string, code: string, job: Job): Promi
   let isESM = false;
 
   try {
-    ast = acorn.parse(code, { ecmaVersion: 2020, allowReturnOutsideFunction: true });
+    ast = acorn.parse(code, { ecmaVersion: 2020, allowReturnOutsideFunction: true, allowAwaitOutsideFunction: true });
```
Comment (guybedford) [inline src/analyze.ts:242]: Actually this is incorrect - CommonJS modules do not support TLA.

```suggestion
    ast = acorn.parse(code, { ecmaVersion: 2020, allowReturnOutsideFunction: true, allowAwaitOutsideFunction: true });
```

**Review by guybedford — CHANGES_REQUESTED** [review]
Let's just remove the CJS TLA case before merging.

**Comment by guybedford** [comment]
We should add this functionality to ncc as well - https://github.com/vercel/webpack-asset-relocator-loader/blob/main/src/asset-relocator.js#L531.

**Review by guybedford — APPROVED** [review]

**Comment by hehex9** [comment]
thanks 😉

> need a maintainer to approve running workflows

**Review by styfle — APPROVED** [review]
Thank you!

**Pull request merged**